src/keysym-utf.c

```diff
@@ -897,6 +897,30 @@ xkb_keysym_to_utf32(xkb_keysym_t keysym)
     return bin_search(keysymtab, ARRAY_SIZE(keysymtab) - 1, keysym);
 }
 
+XKB_EXPORT xkb_keysym_t
+xkb_utf32_to_keysym(uint32_t ucs)
```

Comment:
It'd be easier to verify if the code corresponds exactly to `xkb_keysym_to_utf32`, comments and all, with just the requisite adjustments.

I mean particularly the "patch encoding botch" and "special keysyms" hunks.

Reply:
I have changed the numeric literals to `XKB_KEY_*` constants - hopefully it will make the intention more clear. Since the keypad keysyms have corresponding lower keysyms that produce the same character it will never be returned. Therefore it does not need any special handling like in `xkb_keysym_to_utf32`. The same applies to handling `XKB_KEY_KP_Space` - the `/* patch encoding botch */` is not needed.

Reply:
Thanks.

> Since the keypad keysyms have corresponding lower keysyms that produce the same character it will never be returned. Therefore it does not need any special handling like in xkb_keysym_to_utf32. The same applies to handling XKB_KEY_KP_Space - the /* patch encoding botch */ is not needed.

This will be helpful as a comment in the code, just to save time for anyone comparing the two.